### PR TITLE
Only fire pkg:add messages when the storm package actually changes, resolves SYN-7514

### PR DESCRIPTION
**Pull request by rakuy0**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -2549,7 +2550,12 @@ async def _addStormPkg(self, pkgdef):
         name = pkgdef.get('name')
         olddef = self.pkghive.get(name, None)
         if olddef is not None:
-            await self._dropStormPkg(olddef)
+            if s_hashitem.hashitem(pkgdef) != s_hashitem.hashitem(olddef):
+                await self._dropStormPkg(olddef)
+            else:
+                # if it's the same we can just bounce
+                await self.fire('core:pkg:onload:complete', pkg=name)
```
Comment (Cisphyx) [inline synapse/cortex.py:2557]: Should we actually be firing this even though we didn't run the onload? Also `loadStormPkg` only fires this if there is actually an `onload` defined so we should at least check that

File: synapse/lib/stormsvc.py [hunk 2]
```diff
@@ -139,23 +140,37 @@ async def _runSvcInit(self):
...
+        oldpkgs = self.core.getStormSvcPkgs(self.iden)
+        byname = {}
+        done = set()
+        for pdef in oldpkgs:
+            iden = s_hashitem.hashitem(pdef)
+            byname[pdef.get('name')] = iden
 
         # Register new packages
         for pdef in self.info.get('pkgs', ()):
+            pdef['svciden'] = self.iden
```
Comment (Cisphyx) [inline synapse/lib/stormsvc.py:154]: probably need this to still be in the `try` block to handle malformed pkg definitions

File: synapse/lib/stormsvc.py [hunk 3]
```diff
@@ -165,6 +180,11 @@ async def _runSvcInit(self):
                 name = pdef.get('name')
```
Comment (Cisphyx) [inline synapse/lib/stormsvc.py:180]: already grabbed `name`, can skip getting it again here

File: synapse/cortex.py (hunk 1 above)
Comment (rakuy0) [reply synapse/cortex.py:2557]: Good point. It made sense initially but probably better to not. I'll just remove this.

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**